server/core/src/main/java/com/mongodb/stitch/server/core/auth/internal/StitchAuthImpl.java

```diff
@@ -50,7 +53,17 @@ public StitchAuthImpl(
   }
 
   @Override
-  public <T> T getProviderClient(final AuthProviderClientFactory<T> provider) {
+  public <ClientT> ClientT getAuthenticatedProviderClient(
+          final AuthProviderClientFactory<ClientT, StitchAuthRequestClient> provider) {
+    if (!isLoggedIn()) {
```

Comment:
[minor] We should always let these be created. You may log out while you have one anyway. You don't need an authenticated auth request client to get here anyway.

Reply:
The same change will need to be made on iOS, I'll file a ticket since I don't want to touch iOS until Jason is done with it